src/plugins/controls/server/options_list/options_list_expensive_suggestion_queries.ts

```diff
@@ -226,4 +229,48 @@ const expensiveSuggestionAggSubtypes: { [key: string]: OptionsListSuggestionAggr
       };
     },
   },
+  /**
+   * the "date" query / parser should be used when the options list is built on a field of type date.
+   */
+  date: {
+    buildAggregation: ({ fieldName, sort, size }: OptionsListRequestBody) => {
+      const dateQuery: any = {
```

Comment:
It looks like this is more-or-less identical to `textOrKeywordOrNested`? Could we not just call this for `date` fields as well? So make it `textOrKeywordOrNestedOrDate` (which is definitely a mouthful 😆 )

Reply:
Lol I suppose we could consolidate the two. The main difference is the date agg doesn't support a search string and will error out if we try to do a terms agg with a search string on a date field, so I broke it out into its own separate query, but since the search bar is disabled on the UI side, `searchString` should never actually have a value when building an agg for a date field. Even with the duplication of code, I just though it would be clearer from a readable code standpoint to be explicit about not supporting search strings for the date query.

Reply:
As a sanity check, we could set the search string to empty if `field.type` is date - would that help clarify? Since this is **so** much code, I lean towards not duplicating in this case.